luigi/task.py

```diff
@@ -817,19 +817,16 @@ def getpaths(struct):
     if isinstance(struct, Task):
         return struct.output()
     elif isinstance(struct, dict):
-        r = {}
-        for k, v in six.iteritems(struct):
-            r[k] = getpaths(v)
-        return r
+        return struct.__class__((k, getpaths(v)) for k, v in six.iteritems(struct))
```

Comment:
I believe this would crash if you use a [`collections.Counter`](https://docs.python.org/3.6/library/collections.html#collections.Counter). But if you use that for this purpose, you totally deserve it. ;p

To be clear. This code is better, there's no need for a fix. :)